### PR TITLE
Remove support for streaming in SavedModelRunner

### PR DESCRIPTION
**Pull request by drasmuss**

### CODE REVIEW

File: nengo_edge/tests/test_network_runner.py [hunk 1]
```diff
@@ -1,68 +1,44 @@
 # pylint: disable=missing-docstring
 
+# TODO: rename this to test_coral_runner
```
Comment (jchalatu) [inline nengo_edge/tests/test_network_runner.py:3]: Another TODO here.

**Review by jchalatu — APPROVED** [review]
Changes look good, and the simplifications on the network runners are nice additions.

File: docs/examples/microphone-demo/microphone-demo.ipynb [hunk 2]
```diff
@@ -138,9 +138,9 @@
     "import numpy as np\n",
     "import sounddevice as sd  # pylint: disable=ungrouped-imports\n",
     "\n",
-    "from nengo_edge import SavedModelRunner\n",
+    "from nengo_edge import TFLiteRunner\n",
```
[the remark this answers is not in the document]
Comment (jchalatu) [reply docs/examples/microphone-demo/microphone-demo.ipynb:141]: Ah of course, I forgot that the `TFLiteRunner` was distinct 👍

**Comment by drasmuss** [comment]
Pushed a fixup that moves the non-network NetworkRunner mocking logic from the test suite into the main class (this lets us reuse that logic in `nengo-edge-hw` for the `use_device=False` cases, which is handy).

**Pull request merged**